minichain/p2p.py

```
@@ -101,70 +140,183 @@ def peer_count(self) -> int:
+                peer_addr = data.get("_peer_addr", "")
+                peer_id = (
+                    peer_addr[len("peer:"):] if peer_addr.startswith("peer:") else peer_addr
+                )
+
+                if msg_type not in SUPPORTED_MESSAGE_TYPES:
+                    continue
+                try:
+                    if self._is_duplicate(msg_type, payload):
+                        continue
+                    self._mark_seen(msg_type, payload)
+                except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)